### ISSUE TITLE
Too Many Modded Ores Appears to Result in a Resource Placer Error

### ISSUE THREAD

**Issue by Sederien** [tburrows13/FreightForwarding#12]
So... feel free to consider this issue a non-issue, but it would seem with enough modded ores that FF's Resource-Autoplace can't handle them. This is when using:

All BZ ores + https://mods.factorio.com/mod/IfNickel and https://mods.factorio.com/mod/BrassTacks (and AAI: Industry). Results in the following when creating a new game (on map load):

![image](https://github.com/tburrows13/FreightForwarding/assets/66249229/41c5876b-7c30-42de-82b6-9c533bd21d8a)

Now, if I drop Alumninum and the requisite Foundry from BZ... no issues. If I drop Noble Metals alone... issues. But Noble Metals and Salt being dropped... no issues.

Seems like the solution is just dropping a mod or two, but it also may be due to the current Resource-Autoplace just not handling a certain number of ores if someone uses all BZ + a couple, all of which note compatibility with FF.

For now, I'll just go with dropping Aluminum and Foundry (since I was forgoing the new building in the options anyway), but if you somehow know an easy config update solution for Resource-Autoplace, that'd be great.

**Comment by tburrows13** [comment]
Thanks for the report. It is a strange one and I don’t really know what is causing it. I’m not too concerned though since Nickel and Brass are not currently supported anyway. They probably work (except when used with all of BZ because of this bug) but they haven’t been checked for tech softlocks and they haven’t been given the FF custom ore spawning which the BZ ores have.

Up to @RedRafe if he wants to do proper compatibility for those, and fix this problem at the same time.

**Comment by RedRafe** [comment]
@tburrows13 @Sederien it could be "If I had a Nickel" and "brass tracks" causing issues here because I've never tested those, but I've tested full BZ suite and there were no such issues when loading a new map.

I'll take a closer look

**Comment by Sederien** [comment]
@tburrows13 @RedRafe Certainly. All makes sense. And no requirement that “If I had a Nickel” or “Brass Tacks” actually works with FF, as that’s not your responsibility in the end.

What I will note is that I specifically ran into the error first after successfully loading and using “If I had a Nickel” and “Brass Tacks” in FF along with _most_ of BZ (_EDIT: For clarity, ran into no errors yet_). It was actually only when I added the last of Brevven’s mods (Aluminum and the required Foundry) that I ran into the above error. (Basically, I was adding mods one by one and checking out the Recipe Book / Tech Tree to get a general idea of how recipes were being modified and whether they seemed balanced/fun. This started only with the recommended BZ mods: Noble Metals, Tungsten, and Natural Gas. Then expanded to a few more BZ and Brass Tacks / If I had a Nickel in sequence. So the fact that the game loaded and seemed to run fine with planetfall’s mods, but not with the full suite led me to experiment.)

I tried various combinations of options, but it seems like the existence of enough modded ores simply borks the Resource-Autoplace script. I don’t know the math behind it, but looking through, I assume in layman’s terms I can understand that there are finite number of spaces allotted by the script for starting resource patches and it has some restrictions as to where patches can be placed to keep things from being too messy… so perhaps the area in which it considers starting is set too small?

In any case, as mentioned, not required that this gets fixed. As @tburrows13 mentioned, Nickel and Brass haven’t been checked for soft locks (seems okay at a glance, but I haven’t run through in full!) nor “given the FF custom ore spawning which the BZ ores have” which I was not aware was a thing. Since planetfall notes compatibility with FF on the main Factorio mod site, might be worth reaching out as FF gets more popular. FF’s a great mod. My thanks.

(P.S. While you’re considering, I did not add as it’s alpha-level, but the mod Bismuth seems to be taking the same approach as BZ/Brass/Nickel. Another to note if the error ends up not just being a simple Resource-Autoplace config.Luca change.)

Thanks for the time you two! Awesome mod regardless.

**Comment by Sederien** [comment]
Minor update that, after review of all of the additional resource mods, I decided the most interesting mix for me would be to play with the following mods installed:

- All BZ including Foundry EXCEPT:
> - "Natural Gas" (in retrospect, looking through the recipes, I like the fluid complications in circuits added here and am willing to toss out Natural Gas's early plastic substitute in favor of Salt & Chlorine's later game challenges)
> - "Noble Metals" (listed by Brevven as still in Alpha and subject to change)

- Both planetfall mods: Brass and Nickel

(EDIT: Originally left out "Salt & Chlorine" as I thought enough is already changed to circuits through BZ Silicon and BZ Carbon... and Issue #13 makes the world gen have even more ore types, but Natural Gas has another spawn as well, so it was an easy switch. Still don't need mineable salt even if enforced by FF.)

Note that I have also optionally turned off Flake Graphite (default: on) and left out Diamond mining (default: off) in BZ Carbon.

As before, by leaving out any two resource-adding mods (_**any two**_ whether planetfall's or Brevven's), the game loads and works without issue.

At this point, given the mods I am using, while claimed by planetfall to support FF, isn't officially supported by FF. So I will consider my current install outside of scope of FF and subsequent errors are at my own risk. I'm just a bit of a fan of how well-thought out / natural some of the additions in "Brass Tacks" and "If I Had a Nickel" are:

![image](https://github.com/tburrows13/FreightForwarding/assets/66249229/011d5b04-45c4-4058-8e6e-aae076fabd46)

If perchance you would like me to keep you updated on whether or not things work out, happy to do so. Otherwise, I'll leave this issue open for you to close when you decide upon whether you want to check into more than X mod resources causing a load error as described in the opening post. More than willing to help if I can.

**Comment by RedRafe** [comment]
@Sederien definitely keep us posted if you can :)
I will look into Planetfall's effects with FF too for loading/map-generation, but a keen eye and feedback on how recipes/balancing turn out is always appreciated as we do not always have that much time to look into every detail of each combination of compatible mods (BTW, you can find that list here for future reference: https://github.com/tburrows13/FreightForwarding/blob/master/compatibility/README.md).

AS LONG AS Planetfall's mods are compatible with BZ's (which are rooted into  FF mechanics) I can add compatibility with those as well :) Even tho the compatibility itself shouldn't take long, we've been postponing the resource-autoplace rewrite for so long, that it might be that this is "the issue" that forces me to finally give that bit a well deserved refactor, in which case it would take a little longer to be released 😅

**Comment by RedRafe** [comment]
Also also you may be interested in joining Xorimuth's discord that has a FF section: https://discord.gg/dp6nvha7
